AudioManagerUtilities/src/CAmSocketHandler.cpp

```diff
@@ -320,20 +320,30 @@ am_Error_e CAmSocketHandler::listenToSignals(const std::vector<uint8_t> & listSi
 
       auto actionPoll = [this](const pollfd pollfd, const sh_pollHandle_t, void*)
       {
-          const VectorSignalHandlers_t & signalHandlers = mSignalHandlers;
-          /* We have a valid signal, read the info from the fd */
-          struct signalfd_siginfo info;
-          ssize_t bytes = read(pollfd.fd, &info, sizeof(info));
-          assert(bytes == sizeof(info));
-
-          /* Notify all listeners */
-          for(auto it: signalHandlers)
-          it.callback(it.handle, info, it.userData);
+            const VectorSignalHandlers_t& signalHandlers = mSignalHandlers;
+            /* We have a valid signal, read the info from the fd */
+            struct signalfd_siginfo info;
+            ssize_t bytes = read(pollfd.fd, &info, sizeof(info));
+            if (bytes == sizeof(info))
```

Comment:
According to signalfd manpage the data is locked until all data is read. The data can be not read partly.
--> Refuse change request

Reply:
Fair enough, just wanted you to confirm that.  Since the question came up - I'd ask for a comment to note this, but again you can refuse that request of course.